frontend/scripts/react-components/nav/filters-nav/filters-nav.component.jsx

```diff
@@ -0,0 +1,114 @@
+import ToolSearch from 'react-components/tool/tool-search/tool-search.container';
+import LocaleSelector from 'react-components/nav/locale-selector/locale-selector.container';
+import AdminLevelFilter from 'react-components/nav/filters-nav/admin-level-filter/admin-level-filter.container';
+
+class FiltersNav extends React.PureComponent {
+  static isActiveLink(match, location, link) {
+    const { type, query = {} } = location;
+    const { payload = {}, type: linkType } = link.page;
+    return (
+      type === linkType &&
+      !!(query.state && query.state.isMapVisible) ===
+        !!(payload.query && payload.query.isMapVisible)
+    );
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      menuOpen: false
+    };
+    this.toggleMenu = this.toggleMenu.bind(this);
+    this.renderMenuOpened = this.renderMenuOpened.bind(this);
+    this.renderMenuClosed = this.renderMenuClosed.bind(this);
+  }
+
+  toggleMenu() {
+    this.setState(state => ({ menuOpen: !state.menuOpen }));
+  }
+
+  renderMenuOpened() {
+    const { links } = this.props;
+    return (
+      <React.Fragment>
+        <div className="filters-nav-left-section">
+          <NavLinksList
+            links={links}
+            listClassName="filters-nav-submenu-list"
+            itemClassName="filters-nav-item"
+            linkClassName="filters-nav-link"
+            linkActiveClassName="filters-nav-link -active"
+            isActiveLink={FiltersNav.isActiveLink}
+          />
+        </div>
+        <div className="filters-nav-right-section">
+          <div className="filters-nav-item">
+            <LocaleSelector />
+          </div>
+          <ToolSearch className="filters-nav-item" />
+        </div>
+      </React.Fragment>
+    );
+  }
+
+  renderMenuClosed() {
+    const { selectedContext } = this.props;
+    return (
+      <React.Fragment>
+        <div className="filters-nav-left-section">
+          <ContextSelector className="filters-nav-item" />
+          {selectedContext && (
+            <React.Fragment>
+              <AdminLevelFilter className="filters-nav-item" />
+              <YearsSelector className="filters-nav-item" />
+            </React.Fragment>
+          )}
+        </div>
+        <div className="filters-nav-right-section">
+          {selectedContext && (
+            <React.Fragment>
+              <ResizeBySelector className="filters-nav-item" />
+              <RecolorBySelector className="filters-nav-item" />
+              <ViewSelector className="filters-nav-item" />
+            </React.Fragment>
+          )}
+          <ToolSearch className="filters-nav-item" />
+        </div>
+      </React.Fragment>
+    );
+  }
+
+  render() {
+    const { menuOpen } = this.state;
+    return (
+      <div className="c-filters-nav">
+        <div className="filters-nav-item">
+          <button className={cx('c-burger', { open: menuOpen })} onClick={this.toggleMenu}>
+            <span className="ingredient" />
```

Comment:
these 3 spans confused me

Reply:
You can't really have a 🍔 without at least 3 ingredients. 😛